webknossos-datastore/app/com/scalableminds/webknossos/datastore/services/MeshFileService.scala

```diff
@@ -368,24 +377,38 @@ class MeshFileService @Inject()(config: DataStoreConfig)(implicit ec: ExecutionC
   def readMeshChunkV3(organizationName: String,
                       dataSetName: String,
                       dataLayerName: String,
-                      meshChunkDataRequest: MeshChunkDataRequestV3,
-  ): Fox[(Array[Byte], String)] = {
-    val meshFilePath = dataBaseDir
-      .resolve(organizationName)
-      .resolve(dataSetName)
-      .resolve(dataLayerName)
-      .resolve(meshesDir)
-      .resolve(s"${meshChunkDataRequest.meshFile}.$meshFileExtension")
-
-    executeWithCachedHdf5(meshFilePath, meshFileCache) { cachedMeshFile =>
-      val meshFormat = cachedMeshFile.reader.string().getAttr("/", "mesh_format")
-      val data =
-        cachedMeshFile.reader
-          .uint8()
-          .readArrayBlockWithOffset("neuroglancer", meshChunkDataRequest.byteSize, meshChunkDataRequest.byteOffset)
-      (data, meshFormat)
-    } ?~> "mesh.file.readData.failed"
-  }
+                      meshChunkDataRequests: MeshChunkDataRequestV3List,
+  ): Fox[(Array[Byte], String)] =
+    for {
+      data: List[(Array[Byte], String, Int)] <- Fox.combined(
```

Comment:
Could you test if Fox.serialCombined slows this down significantly on your machine? The current implementation with Fox.combined and map may parallelize over all entries, which could actually be slower on a spinning disk.

Reply:
Thanks for looking into this! Yes, I’d say for the moment we should optimize for spinning disk (which is also what the sorting-by-byte-offset step is for)